public/js/roles/roles-list.controller.js

```diff
@@ -64,5 +65,30 @@
                 vm.getRoles(vm.page);
             });
         }
+
+        function deleteRole (roleId) {
+            api.getUsers().then(function (res) {
+                const usersHasRole = res.data.filter(e => e.roles.includes(roleId));
+                const targetRole = vm.items.filter(e => { return e._id === roleId; })[0];
```

Comment:
If you need to find the first element satisfying a condition, use [`Array.prototype.find`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of `filter`

Or, in this case, you can pass the full role object (instead of its id) to deleteRole when it's called in the template. That would avoid an unnecessary search.

Reply:
I change it by find method :)